raiden/network/transport/matrix/transport.py

```diff
@@ -273,40 +274,29 @@ def __init__(self, config: dict):
         self._config = config
         self._raiden_service: RaidenService = None
 
+        if config['server'] == 'auto':
+            available_servers = config['available_servers']
+        elif urlparse(config['server']).scheme in {'http', 'https'}:
+            available_servers = [config['server']]
+        else:
+            raise TransportError('Invalid matrix server specified (valid values: "auto" or a URL)')
+
         def _http_retry_delay() -> Iterable[float]:
```

Comment:
Shouldn’t this be part of the utility?

Reply:
@ulope this had too much to do with internals and variables of `config`, and I wanted to make the utility function as agnostic in relation to config as possible, that's why I kept this outside of it, but can be as easily used outside raiden as any other utility function, so we can say it's _already split_